From e48f59d2b9bcb43ea8a7b8c884dcb73c8f65b170 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 19 Oct 2009 14:27:20 +1000
Subject: [PATCH] Un-ignore absolute axes for the Xen Virtual Pointer.

The evdev-internal axis type picking fails for the Xen Virtual Pointer as it
exposes both relative and absolute axes. Evdev picks the relative axes by
default, leading to a immovable pointer if the Xen backend only sends
absolute coordinates.

Explicitly tell evdev to not ignore the absolute axes.

https://bugzilla.redhat.com/show_bug.cgi?id=523914

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 fdi/policy/10osvendor/10-x11-input.fdi |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/fdi/policy/10osvendor/10-x11-input.fdi b/fdi/policy/10osvendor/10-x11-input.fdi
index 89ba672..752c4ab 100644
--- a/fdi/policy/10osvendor/10-x11-input.fdi
+++ b/fdi/policy/10osvendor/10-x11-input.fdi
@@ -44,5 +44,12 @@
     <match key="info.product" contains="ThinkPad HDAPS accelerometer data">
         <remove key="input.x11_driver" />
     </match>
+
+    <!-- https://bugzilla.redhat.com/show_bug.cgi?id=523914
+         Mouse does not move in PV Xen guest -->
+    <match key="info.product" contains="Xen Virtual Pointer">
+      <merge key="input.x11_options.IgnoreAbsoluteAxes" type="string">false</merge>
+      <merge key="input.x11_options.IgnoreRelativeAxes" type="string">false</merge>
+    </match>
   </device>
 </deviceinfo>
-- 
1.6.2.5

